### PR TITLE
Fix issue #478

### PR DESCRIPTION
**Pull request by daminichopra**
## Proposed changes

Return a values field in /entities containing the last value

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [ x ] I have read the [CONTRIBUTING][contrib] doc
- [ x ] I have signed the [CLA][cla]
- [ x ] I have added tests that prove my fix is effective or that my
      feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments
Fix for [Issue](https://github.com/orchestracities/ngsi-timeseries-api/issues/478)

[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual-cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/smartsdk/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"

### CODE REVIEW

**Comment by daminichopra** [comment]
Gentle Reminder!!

File: src/translators/sql_translator.py [hunk 1]
```diff
@@ -1158,59 +1158,48 @@ def query_ids(self,
...
-        stmt = ''
         if len(table_names) > 0:
             for tn in sorted(table_names):
                 len_tn += 1
                 if len_tn != len(table_names):
-                    stmt += "select " \
-                            "entity_id, " \
-                            "entity_type, " \
-                            "max(time_index) as time_index " \
+                    op = "select * " \
```
Comment (chicco785) [inline src/translators/sql_translator.py:1169]: the original query composed results from different tables, this is not the case anymore.

File: docs/manuals/user/index.md [hunk 2]
```diff
@@ -1,456 +1,324 @@
...
-  (See [PR 191](https://github.com/orchestracities/ngsi-timeseries-api/pull/191)),
-  The following limitations still apply: a error in a single data entity will invalidate
-  the all set. There is not optimisation for large message size.
-
-- Data are assumed to be consistent. I.e., if the first data notification for
-  an entity type use a given set of data types for the attributes, the following
-  data notifications must be consistent, or they will be rejected. E.g. if
-  the data type of attribute `speed` of entity type `car` is set initially
-  to `Number`, later on it cannot be set to `Text`.
+# QuantumLeap
```
Comment (chicco785) [inline docs/manuals/user/index.md:1]: not part of this pr, should be removed

File: src/reporter/delete.py [hunk 3]
```diff
@@ -34,11 +34,19 @@ def delete_entity(entity_id, type_=None, from_date=None, to_date=None):
 def delete_entities(entity_type, from_date=None, to_date=None,
                     drop_table=False):
     with translator_for(fiware_s()) as trans:
-        if drop_table:
-            trans.drop_table(etype=entity_type, fiware_service=fiware_s())
-            logging.getLogger(__name__).info(
-                "dropped entity_type {}".format(entity_type))
-            return 'entity table dropped', 204
+        if fiware_sp() is not None:
```
Comment (chicco785) [inline src/reporter/delete.py:37]: not part of this pr, should be removed

**Review by chicco785 — CHANGES_REQUESTED** [review]

**Comment by chicco785** [comment]
this pr needs a rebase, but due to other changes operated in the code (see the pr from your colleague: https://github.com/orchestracities/ngsi-timeseries-api/pull/486) there will be conflicts to be solved.

File: src/translators/sql_translator.py (hunk 1 above)
Comment (daminichopra) [reply src/translators/sql_translator.py:1169]: > the original query composed results from different tables, this is not the case anymore.

@chicco785 , Yes .. modified as per this [issue](https://github.com/orchestracities/ngsi-timeseries-api/issues/478#issuecomment-874471676)

**Comment by daminichopra** [comment]
> this pr needs a rebase, but due to other changes operated in the code (see the pr from your colleague: #486) there will be conflicts to be manually solved.

@chicco785 , I have updated files. Also, please let me know if i missed something. Thank you :)

File: src/translators/sql_translator.py (hunk 1 above)
Comment (chicco785) [reply src/translators/sql_translator.py:1169]: that comments is about enabling the functionality for single entity, not about giving up the listing of entities as it is now.

Comment (chicco785) [reply src/translators/sql_translator.py:1169]: also the missing entitytype for the other queries, was already solved by #486

**Comment by chicco785** [comment]
> > this pr needs a rebase, but due to other changes operated in the code (see the pr from your colleague: #486) there will be conflicts to be manually solved.
> 
> @chicco785 , I have updated files. Also, please let me know if i missed something. Thank you :)

as mentioned the union query is required, i fixed the code accordingly.

**Review by chicco785 — APPROVED** [review]

**Comment by chicco785** [comment]
given that @pooja1pathak is working on setting up as "a datasource" endpoint, i will return to the basic version that returns only the last index.
